### PR TITLE
Create UI to improve segmentation

### PR DESCRIPTION
**Pull request by arinkulshi-skylight**
# Pull Request

## Description
Create UI to ease the pain of segmenting a PDF 

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: OCR/ocr/services/segmentation_ui.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    QInputDialog,
+    QGraphicsView,
+    QGraphicsScene,
+    QGraphicsRectItem,
+)
+from PyQt5.QtGui import QPixmap, QPainter, QColor, QPen, QBrush, QKeyEvent
+from PyQt5.QtCore import Qt, QRectF
+
+
+class ImageViewer(QGraphicsView):
```
[the remark this answers is not in the document]
Comment (arinkulshi-skylight) [reply OCR/ocr/services/segmentation_ui.py:22]: Good point I have added a docstring to the top level class.

File: OCR/ocr/services/segmentation_ui.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            if event.angleDelta().y() > 0:
+                self.scale(zoom_factor, zoom_factor)
+            else:
+                self.scale(1 / zoom_factor, 1 / zoom_factor)
+        else:
+            super().wheelEvent(event)
+
+    def keyPressEvent(self, event: QKeyEvent):
+        # hotkeys for zooming and scrolling
+        if event.modifiers() & Qt.ControlModifier:
```
[the remark this answers is not in the document]
Comment (arinkulshi-skylight) [reply OCR/ocr/services/segmentation_ui.py]: Good idea. I have made this match and added the cases.

**Review by knguyenrise8 — APPROVED** [review]
